core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -659,10 +669,8 @@ final class FiberRuntime[E, A](fiberId: FiberId.Runtime, fiberRefs0: FiberRefs,
 
   private[zio] def getStatus(): Fiber.Status =
     if (_exitValue ne null) Fiber.Status.Done
-    else {
-      if (_asyncContWith ne null) Fiber.Status.Suspended(self._runtimeFlags, _lastTrace, _blockingOn())
-      else Fiber.Status.Running(self._runtimeFlags, _lastTrace)
-    }
+    else if (running.get()) Fiber.Status.Running(self._runtimeFlags, _lastTrace)
```

Comment:
`_asyncContWith` is _not_ volatile, we cannot use it to accurately determine the status of a fiber. However, we've been using it in `TestClock` extensively to poll for the status of other fibers, so we need to use a volatile var.

`running` is not great either; it doesn't mean the fiber is actually running; more that it's been "un-suspended". I had to do this change as while I was working on this PR I had tests misbehaving and this change fixed the issue.

I think this behaviour is more correct but if anyone disagrees let me know :)